### PR TITLE
[kafka] limit when `messaging.kafka.message.key` attribute is set to basic types

### PR DESCRIPTION
**Pull request by lachmatt**
## Why

Fixes #

## What

Adjust to better align with semantic conventions [recommendation](https://github.com/open-telemetry/semantic-conventions/blob/ac9f85f0581e1fd621f539392a7218c5cf4f3a25/docs/messaging/kafka.md?plain=1#L38)

## Tests

Added in PR.

## Checklist

~~- [ ] `CHANGELOG.md` is updated.~~
~~- [ ] Documentation is updated.~~
~~- [ ] New features are covered by tests.~~

### CODE REVIEW

File: src/OpenTelemetry.AutoInstrumentation/Instrumentations/Kafka/KafkaInstrumentation.cs [hunk 1]
```diff
@@ -116,6 +116,16 @@ public static void SetDeliveryResults(Activity activity, IDeliveryResult deliver
             deliveryResult.Offset.Value);
     }
 
+    internal static string? ExtractMessageKeyValue(object key)
+    {
+        return key switch
+        {
+            string s => s,
+            int or long or float or double => key.ToString(),
```
Comment (Kielek) [inline src/OpenTelemetry.AutoInstrumentation/Instrumentations/Kafka/KafkaInstrumentation.cs:124]: InvariantCulture?

Comment (lachmatt) [reply src/OpenTelemetry.AutoInstrumentation/Instrumentations/Kafka/KafkaInstrumentation.cs]: https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/3379/commits/bb08fdeec91e5e63bcb93ca65023e20147ed2ff0

File: src/OpenTelemetry.AutoInstrumentation/Instrumentations/Kafka/KafkaInstrumentation.cs [hunk 2]
```diff
@@ -116,6 +117,16 @@ public static void SetDeliveryResults(Activity activity, IDeliveryResult deliver
             deliveryResult.Offset.Value);
     }
 
+    internal static string? ExtractMessageKeyValue(object key)
+    {
+        return key switch
+        {
+            string s => s,
+            int or long or float or double => Convert.ToString(key, CultureInfo.InvariantCulture),
```
Comment (pellared) [inline src/OpenTelemetry.AutoInstrumentation/Instrumentations/Kafka/KafkaInstrumentation.cs:125]: I suggest also adding:
- `uint`
- `ulong`
- `decimal`

(even if they are not yet handled by the instrumented library

**Review by pellared — APPROVED** [review]

File: src/OpenTelemetry.AutoInstrumentation/Instrumentations/Kafka/KafkaInstrumentation.cs (hunk 2 above)
Comment (Kielek) [reply src/OpenTelemetry.AutoInstrumentation/Instrumentations/Kafka/KafkaInstrumentation.cs:125]: byte, short, etc ; )

**Pull request merged**